src/test/java/org/folio/circulation/domain/OverdueFineCalculatorServiceTest.java

```diff
@@ -211,19 +210,22 @@ public void shouldNotCreateFeeFineRecordWhenItemDoesNotExist()
     Loan loan = createLoan();
 
     when(overdueFinePolicyRepository.findOverdueFinePolicyForLoan(any()))
-      .thenReturn(CompletableFuture.completedFuture(Result.succeeded(loan)));
+      .thenReturn(completedFuture(succeeded(loan)));
     when(overduePeriodCalculatorService.getMinutes(any(), any()))
-      .thenReturn(CompletableFuture.completedFuture(Result.succeeded(5)));
+      .thenReturn(completedFuture(succeeded(5)));
     when(itemRepository.fetchItemRelatedRecords(any()))
-      .thenReturn(CompletableFuture.completedFuture(Result.succeeded(null)));
+      .thenReturn(completedFuture(Result.succeeded(null)));
+    when(feeFineOwnerRepository.getFeeFineOwner(SERVICE_POINT_ID.toString()))
+      .thenReturn(completedFuture(succeeded(createFeeFineOwner())));
+    when(feeFineRepository.getFeeFine(FEE_FINE_TYPE, true))
+      .thenReturn(completedFuture(succeeded(createFeeFine())));
 
     CheckInProcessRecords records = new CheckInProcessRecords(
       CheckInByBarcodeRequest.from(createCheckInByBarcodeRequest()).value())
       .withLoan(loan);
 
     overdueFineCalculatorService.calculateOverdueFine(records).get();
     verifyNoInteractions(feeFineOwnerRepository);
-    verifyNoInteractions(feeFineRepository);
```

Comment:
Why do we no longer need these checks?

Reply:
I changed the order of fetching required entities, so that fee/fine type is fetched first. I did that because if the type is missing, there is no point to make the calls (many) for an item and item-related records. Just a little performance optimization.